src/cpp/src/icontinuous_batching.cpp

```diff
@@ -35,47 +35,74 @@ ContinuousBatchingPipeline::IContinuousBatchingPipeline::generate(
         // ov::genai::add_special_tokens(false) is aligned with stateful pipeline
         input_ids.push_back(m_tokenizer.encode(history, ov::genai::add_special_tokens(false)).input_ids);
+        tokenization_durations.emplace_back(PerfMetrics::get_microsec(std::chrono::steady_clock::now() - encode_start));
         timer.end();
     } else {
         input_ids.reserve(prompts.size());
         timer.start();
         for (const std::string& prompt : prompts) {
+            const auto encode_start = std::chrono::steady_clock::now();
             input_ids.push_back(m_tokenizer.encode(prompt).input_ids);
+            tokenization_durations.emplace_back(PerfMetrics::get_microsec(std::chrono::steady_clock::now() - encode_start));
         }
         timer.end();
     }
 
     std::vector<EncodedGenerationResult> encoded = generate(input_ids, sampling_params, streamer);
+    
+    // auto& raw_counters = perf_metrics.raw_metrics;
+    // For all encoded results perf_metrics are the same except for tokenization time.
+    // raw_counters.generate_durations = std::vector<MicroSeconds>();
+    // raw_counters.generate_durations.emplace_back(PerfMetrics::get_microsec(stop_time - start_time));
+    // raw_counters.tokenization_durations.emplace_back(PerfMetrics::get_microsec(decode_start_time - start_time));
+    // raw_counters.detokenization_durations.emplace_back(PerfMetrics::get_microsec(stop_time - decode_start_time));
+    // perf_metrics.m_evaluated = false;
+    // perf_metrics.evaluate_statistics(start_time);
 
     std::vector<GenerationResult> decoded;
+    auto decode_start_time =  std::chrono::steady_clock::now();
     decoded.reserve(encoded.size());
-    for (EncodedGenerationResult& res : encoded) {
+    for (size_t i = 0; i < encoded.size(); ++i) {
+        EncodedGenerationResult res = encoded[i];
+        auto& perf_metrics = res.perf_metrics;
+        auto& raw_counters = perf_metrics.raw_metrics;
+        raw_counters.tokenization_durations.emplace_back(tokenization_durations[i]);
+
         std::vector<std::string> generated;
         generated.reserve(res.m_generation_ids.size());
         for (size_t idx = 0; idx < res.m_generation_ids.size(); ++idx) {
+            const auto decode_start = std::chrono::steady_clock::now();
             generated.push_back(m_tokenizer.decode(res.m_generation_ids.at(idx)));
+            raw_counters.detokenization_durations.emplace_back(std::chrono::steady_clock::now() - decode_start);
             if (m_is_chat_conversation && 0 == idx) {
                 m_history.push_back({{"role", "assistant"}, {"content", generated.back()}});
             }
         }
 
+
         decoded.push_back(GenerationResult{
             res.m_request_id,
             std::move(generated),
             std::move(res.m_scores),
-            res.m_status
+            res.m_status,
+            perf_metrics,
         });
     }
+    auto stop_time = std::chrono::steady_clock::now();
```

Comment:
not used currently, but I think you need to fill in `generate_durations` for `ContinuousBatchingPipeline` as well?

Reply:
Removed this stop_time. Agreed that we need to return evaluated metrics with valid generate_durations. Done that